src/main/sensors/battery.c

```diff
@@ -187,10 +190,10 @@ void batteryUpdatePresence(void)
         batteryCriticalVoltage = batteryCellCount * batteryConfig()->vbatmincellvoltage;
     } else if (
         voltageState != BATTERY_NOT_PRESENT
-        && voltageMeter.filtered <= batteryConfig()->batteryNotPresentLevel
         && isVoltageStable
+        && !isVoltageFromBat
```

Comment:
I did test when it went from the 2S+ range to 0V and I didn't notice any issues, but I wasn't looking specifically at that.  The isVoltageStable should catch that and not allow the disconnect until it settles (isVoltageStable)